dz15/src/app.js

```diff
@@ -0,0 +1,127 @@
+async function getInfoFromDummyjsonProductsCategories() {
+  const response = await fetch('https://dummyjson.com/products/categories')
+  const data = response.json()
+  return data
+}
+
+async function getCountFromDummyjsonProductsCategories() {
+  const counter = await getInfoFromDummyjsonProductsCategories()
+  return counter.length
+}
+
+async function findCategories(data, categories) {
+  const result = data.find((category) => category.name === categories)
+  if (result !== undefined) {
+    return result
+  }
+  return 0
+}
+
+async function listOfProductsFromCategory(categories) {
+  const trueCategory = categories.filter((category) => category !== 0)
+  const listOfProducts = await Promise.all(trueCategory.map(
+    async (category) => {
+      const response = await fetch(category.url)
+      const data = await response.json()
+      return data.products.map((element) => ({
+        id: element.id,
+        title: element.title
+      }))
+    }
+  ))
+
+  return listOfProducts
+}
```

Comment:
```suggestion
function getProductsByCategories(categories) {
  return Promise.all(
    categories.filter((category) => category).map(
      async ({ url }) => {
        const response = await fetch(url)
        const { products } = await response.json()

        return products.map(({ id, title }) => ({
          id,
          title
        }))
      }
    )
  )
}
```

занятная функция, немного не по заданию, ну да ладно. держи вариант ещё короче 😅

Reply:
краткость и читаемость становятся антагонистами)